### PR TITLE
Upgrade to be able to handle svelte 5 features

### PR DESCRIPTION
**Pull request by ElijahJohnson5**
Add module js and ts support
Update tests for new warnings
Add tests for module js and ts support
Update dependency to svelte 5

I am open to any changes! Hope this helps get the svelte 5 support across the line.

### CODE REVIEW

**Comment by 03juan** [comment]
Closes #250

**Comment by ravides19** [comment]
Waiting for this PR to be merged 💯

File: index.ts [hunk 1]
```diff
@@ -36,7 +44,7 @@ interface esbuildSvelteOptions {
      * A function to filter out warnings
      * Defaults to a constant function that returns `true`
      */
-    filterWarnings?: (warning: Warning) => boolean;
+    filterWarnings?: (warning: Warning) => warning is Warning;
```
Comment (EMH333) [inline index.ts:47]: Why does this return value need to change?

File: index.ts [hunk 2]
```diff
@@ -155,6 +163,69 @@ export default function sveltePlugin(options?: esbuildSvelteOptions): Plugin {
                 let originalSource = await promisify(readFile)(args.path, "utf8");
                 let filename = relative(process.cwd(), args.path);
 
+                let source = originalSource;
+
+                if (filename.endsWith(".svelte.ts")) {
```
Comment (EMH333) [inline index.ts:168]: placeholder for me to investigate: does Svelte 5 really need this compilation of TS files at this stage?

File: index.ts [hunk 3]
```diff
@@ -1,20 +1,28 @@
...
 interface esbuildSvelteOptions {
     /**
      * Svelte compiler options
      */
     compilerOptions?: CompileOptions;
 
+    /**
+     * Svelte compiler options for module files (*.svelte.js and *.svelte.ts)
+     */
+
```
Comment (EMH333) [inline index.ts:23]: Nit: extra line here so it isn't getting picked up in the `.d.ts` file

**Review by EMH333 — COMMENTED** [review]
This is a great start!

Test failure in Node 16 is due to Svelte 5 bumping minimum version to Node 18. I think this is fine to follow, even if the release will still support Svelte `>=4.2.1` since v16 has been end of life for a while now. Folks can still use older versions of esbuild-svelte if they need compatibility.

File: index.ts (hunk 3 above)
Comment (ElijahJohnson5) [reply index.ts]: Fixed!

File: index.ts (hunk 1 above)
Comment (ElijahJohnson5) [reply index.ts]: It doesn't my bad!

File: index.ts [hunk 4]
```diff
@@ -90,7 +98,7 @@ const shouldCache = (
     },
 ) => build.initialOptions?.incremental || build.initialOptions?.watch;
 
-const SVELTE_FILTER = /\.svelte$/;
+const SVELTE_FILTER = /(\.svelte$)|(\.svelte.js$)|(\.svelte.ts$)/;
```
[the remark this answers is not in the document]
Comment (ElijahJohnson5) [reply index.ts]: I separated them into separate variables, let me know if that looks good to you

File: index.ts (hunk 2 above)
Comment (EMH333) [reply index.ts]: yes. yes it does :(

**Pull request merged**